store/store.go

```diff
@@ -549,10 +550,23 @@ func (s *Store) RemoveACI(key string) error {
 		}
 
 		uid := os.Getuid()
-		dirUid := int(fi.Sys().(*syscall.Stat_t).Uid)
+		dirGid := int(fi.Sys().(*syscall.Stat_t).Gid)
 
-		if uid != dirUid && uid != 0 {
-			return fmt.Errorf("permission denied on %s, directory owner id (%d) does not match current real id (%d). Are you root or the owner of the image?", path, dirUid, uid)
+		gids, err := os.Getgroups()
+		if err != nil {
+			return errwrap.Wrap(errors.New("error getting our own groups"), err)
+		}
+		gids = append(gids, os.Getgid())
+
+		userInDirGroup := false
+		for _, g := range gids {
+			if g == dirGid {
+				userInDirGroup = true
+			}
+		}
+
+		if !userInDirGroup && uid != 0 {
+			return fmt.Errorf("permission denied on %s, user not in the directory group id (%d). Are you root or in the %q group?", path, dirGid, common.RktGroup)
 		}
```

Comment:
I think that, if removed, and permissions are correctly setup it should return an error anyway right?
The check is useful just for better error reporting.
 
Probably this check, if kept, should be used also in WriteACI, ReadStream etc... (or are there upper checks for this?)